Jetpack Plugin: Move <LoadingCard> to be generally accessible

kallehauge opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
### Introduction

This PR aims to make the simple `<LoadingCard>` component easier accessible by other components in the Jetpack Dashboard.


* Prerequisite for #21196

### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

* Move `<LoadingCard>` to the `/components/` directory.
* Rename it to `<JetpackLoadingIcon>` so it doesn't clash with other instances in the code where there is "local versions" with the same name.

### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

There is no direct discussion but it's part of splitting up #21196 which is part of a marketing effort that can be found here: p1HpG7-cY7-p2 

### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

Nope!

### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Spin up a version of this branch.
* Go to `/wp-admin/admin.php?page=jetpack#/recommendations/summary`.
* Validate that you see a blue Jetpack logo (see screenshot below).
* Go to `/wp-admin/admin.php?page=jetpack#/recommendations/product-suggestions`.
* Validate that you see a blue Jetpack logo (see screenshot below).

### Screenshots

`/wp-admin/admin.php?page=jetpack#/recommendations/summary`:

![Screenshot 2021-09-27 at 18 14 59](https://user-images.githubusercontent.com/3846700/134946702-b145f0cb-4996-44e8-a07a-d4803e85298c.png)

`/wp-admin/admin.php?page=jetpack#/recommendations/product-suggestions`:

![Screenshot 2021-09-27 at 18 15 50](https://user-images.githubusercontent.com/3846700/134946805-543ebad4-d9af-4215-888e-5bf4b8a27dd4.png)

kallehauge commented:
Aside: I'm currently preparing tests for this code as well but it shouldn't change much in terms of functional/technical review 🙂

IanRamosC commented on projects/plugins/jetpack/_inc/client/components/jetpack-loading-icon/index.jsx:
```diff
@@ -13,10 +13,10 @@ import { imagePath } from 'constants/urls';
-		<div className="jp-recommendations-loading-card">
-			<img src={ imagePath + '/jetpack-logomark-blue.svg' } alt="Loading recommendations" />
+		<div className="jp-loading-icon">
+			<img src={ imagePath + '/jetpack-logomark-blue.svg' } alt="Loading..." />
```
I think it would be cool to get the `alt` text as a prop, too. So we can use custom alt texts for different contexts

kallehauge replied:
Yes! Great catch! 👏

IanRamosC reviewed (approved):
Code loooks good. Just left a small suggestion that doesn't need to be implemented right now

jeherve reviewed (commented):
This makes sense, although I wonder if we should take that opportunity to move it to the `jetpack/components` package ([#](https://github.com/Automattic/jetpack/tree/177f6496fc7f8f92294e483876310f57124ab331/projects/js-packages/components/components)), where it will be available to other Jetpack plugins as well.

@sergeymitr What's your take on this?

kallehauge commented:
@jeherve I definitely think a loading state/component would make sense as a RNA component! Although, I think design would have to approve this particular one before it goes in as a RNA component since the logo is only referenced as "green" in the Design Handbook.
We could add a style/color/type property to the component but I'm not sure if we ever want a "blue" reference in RNA.

In the final RNA version I would love to see it in a less static version as well :b Like; making it into a spinner by having a line going around the logo in the outer border or something to indicate a loading state. But this is moving into design feedback and not architecture 😅

IanRamosC reviewed (dismissed):
This looks good from my side

kallehauge commented:
Great! Thanks guys! I've rebased the branch and corrected the ellipsis lint error.
Auto-merge has also been enabled, so as soon as the requested change has been approved it should slide right in 😄
